### PR TITLE
Feature/unit tests

### PR DESCRIPTION
**Pull request by bpauli**
add Jest based unit tests for dataLayer

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,6 +20,12 @@ Then choose from the following gulp tasks:
 * `gulp` - generates the build in the `./dist` folder and runs a development server on `localhost:3000`.
 * `gulp build` - generates the build in the `./dist` folder.
 
+### Run unit tests
+To run the unit tests execute the command:
+```
+npm test
```
Comment (richardhand) [inline README.md]: @pauli, can we run them via a gulp task? `gulp test`, that way we can also include it in our build tasks.

**Comment by jckautzmann** [comment]
Would it make sense to have the unit tests in a separate folder? Something like:
- for js sources: `src/main/scripts/datalayer.js`
- for unit tests: `src/test/scripts/datalayertest.js`

**Comment by jckautzmann** [comment]
We should have unit tests for all the methods.
Is there a way to measure the code coverage of the tests?
We might want to define some threshold in the build process.

File: README.md (hunk 1 above)
Comment (richardhand) [reply README.md]: This looks like it can help us out https://www.npmjs.com/package/gulp-jest

**Comment by richardhand** [comment]
Agree that it makes sense to put the tests in a separate folder, but maybe something like this?
```
src/scripts/**/*.js
src/tests/**/*.js
```

**Pull request merged**